azuredevops/resource_commit.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredevops
+
+import (
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/git"
+)
+
+func newCommit(c *CommitClient, commit *git.GitPush) *commitType {
+	return &commitType{
+		k: *commit,
+		c: c,
+	}
+}
+
+var _ gitprovider.Commit = &commitType{}
+
+type commitType struct {
```

Comment:
I don't know about the name, but this should represent a single commit I think.

Reply:
let me know if this is what you refer to

Reply:
would that be what is happening from line 46? 

`for _, commit := range *apiObj.Commits {`

Reply:
as per our meeting i've updated this